exotica_core/src/problems/unconstrained_end_pose_problem.cpp

```diff
@@ -93,18 +93,24 @@ Eigen::RowVectorXd UnconstrainedEndPoseProblem::GetScalarJacobian() const
     return cost.jacobian.transpose() * cost.S * cost.ydiff * 2.0;
 }
 
-double UnconstrainedEndPoseProblem::GetScalarTaskCost(const std::string& task_name) const
+Eigen::VectorXd UnconstrainedEndPoseProblem::GetTaskError(const std::string& task_name) const
```

Comment:
Vlad and I had more discussions earlier on what would be the best way going forward with this - and here is what we decided upon:

1. Please place the `GetTaskError` method into `EndPoseTask` in `tasks.h`.
2. The tasks (`cost`, `equality`, `inequality`) of the corresponding problems should already be exposed as public. As such you can get your intended behaviour by calling `problem.cost.GetTaskError("my_task")`.
3. Do not move the `GetScalarTaskCost` in `UnconstrainedEndPoseProblem`, we will get to that at a later stage.

Background is that for the kinematic `TimeIndexedProblem`s, I did a clean-up in the past moving `GetGoal`, `SetGoal` etc. down into the tasks and using forwarding rather than duplicating the correct indexing in every problem. Mid-term - but not now - we want to move more that is task-related into those rather than the problems. As such, we want to introduce new functionality that is task-related always at the correct place.

Reply:
Makes sense. I moved the method to the task.
There is also a lot of duplication in the derived problems. Most problems are derived directly from the `PlanningProblem`. E.g. the `EndPoseProblem` and `UnconstrainedEndPoseProblem` share many common attributes and methods (`SetGoal`, `GetScalarCost`, `GetRho`, ...). From my point of view, the `EndPoseProblem` is a specialisation of the `UnconstrainedEndPoseProblem` since it adds constraints. Maybe `EndPoseProblem` should be renamed to `ConstrainedEndPoseProblem` and be derived from the `UnconstrainedEndPoseProblem`.